### PR TITLE
Add alerting tools IT; fix missing system index bug of SearchMonitorsTool

### PR DESCRIPTION
**Pull request by ohltyler**
### Description
This PR adds improvements to the existing SearchAlertsTool and SearchMonitors tools:
- adds basic integ tests (more to be added tracked in followup issue #136)
- handles edge cases of missing system indices for SearchMonitorsTool
- cleans up processing logic of responses into helper fns for SearchMonitorsTool
 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ohltyler** [comment]
Note these integ tests will fail until https://github.com/opensearch-project/alerting/pull/1382 is merged and published to sonatype, so the CI can pick up the updated alerting dependency.

**Comment by ohltyler** [comment]
TODOs: add/improve integ tests, ensure everything passes after alerting PR merged.

Meanwhile a first round of review can help speed up merging.

File: src/test/java/org/opensearch/integTest/SearchMonitorsToolIT.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        String agentId = createAgent(registerAgentRequestBody);
+        String agentInput = "{\"parameters\":{\"monitorName\": \"" + monitorName + "\"}}";
+        String result = executeAgent(agentId, agentInput);
+        assertEquals("Monitors=[]TotalMonitors=0", result);
+    }
+
+    // TODO: Add IT to test against sample monitor data
+
+    @SneakyThrows
+    private void createMonitorsSystemIndex(String monitorId, String monitorName) {
```
Comment (lezzago) [inline src/test/java/org/opensearch/integTest/SearchMonitorsToolIT.java:63]: We are trying to create the system index for alerting. The best way to do this is to create a dummy monitor, so that the schema mapping of the index is setup correctly and would be better to test for an integration test.

Comment (ohltyler) [reply src/test/java/org/opensearch/integTest/SearchMonitorsToolIT.java]: Agreed - this was an interim test fn that is actually currently not used. I'll remove and leave the TODO, as this will be grouped with the added tests in the followup issue #136. Thanks for the suggestion, I will likely create dummy monitors and alerts for testing the populated results of the search monitors and search alerts tools respectively.

**Review by lezzago — APPROVED** [review]

**Review by jngz-es — APPROVED** [review]
